src/core.rs

```diff
@@ -250,146 +273,183 @@ impl Core {
-                                                Err(RoutingError::Interface(err)) => Err(err),
-                                                Err(_err) => Ok(()),
-                                                Ok(()) => Ok(()),
-                                            })
-                                            .is_err() {
-                                    return;
-                                }
-                            }
-                            Action::ClientSendRequest { content, dst, result_tx, } => {
-                                if result_tx.send(if let Ok(src) = self.get_client_authority() {
-                                                let request_msg = RequestMessage {
-                                                    content: content,
-                                                    src: src,
-                                                    dst: dst,
-                                                };
-
-                                                match self.send_request(request_msg) {
-                                                    Err(RoutingError::Interface(err)) => Err(err),
-                                                    Err(_err) => Ok(()),
-                                                    Ok(()) => Ok(()),
-                                                }
-                                            } else {
-                                                Err(InterfaceError::NotConnected)
-                                            })
-                                            .is_err() {
-                                    return;
-                                }
-                            }
-                            Action::CloseGroup { name, result_tx, } => {
-                                let close_group = self.routing_table
-                                                      .close_nodes(&name)
-                                                      .map(|infos| {
-                                                          infos.iter()
-                                                               .map(NodeInfo::name)
-                                                               .cloned()
-                                                               .collect()
-                                                      });
-
-                                if result_tx.send(close_group).is_err() {
-                                    return;
-                                }
-                            }
+        (action_sender, core)
+    }
 
-                            Action::Name { result_tx, } => {
-                                if result_tx.send(*self.name()).is_err() {
-                                    return;
-                                }
-                            }
+    /// If there is an event in the queue, processes it and returns true.
+    /// otherwise returns false. Never blocks.
+    #[allow(unused)]
```

Comment:
Why don't you use `#[cfg(feature = "use-mock-crust")]` here? (Similarly in `run`.)